src/workerd/api/hibernatable-web-socket.h

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2017-2022 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+#pragma once
+
+#include <kj/debug.h>
+
+#include <workerd/io/worker-interface.capnp.h>
+#include <workerd/api/global-scope.h>
+#include <workerd/io/worker-interface.h>
+
+namespace workerd::api {
+
+class HibernatableWebSocketMessageEvent final: public ExtendableEvent {
+public:
+  explicit HibernatableWebSocketMessageEvent(v8::Isolate* isolate, rpc::EventDispatcher::HibernatableWebSocketMessageParams::Reader params);
+
+  static jsg::Ref<HibernatableWebSocketMessageEvent> constructor(kj::String type) = delete;
+
+  JSG_RESOURCE_TYPE(HibernatableWebSocketMessageEvent) {
+      JSG_INHERIT(ExtendableEvent);
+  }
+};
+
+class HibernatableWebSocketMessageController final: public jsg::Object {
+  // Type used when calling a module-exported HibernatableWebSocketMessage event handler.
+public:
+  HibernatableWebSocketMessageController(jsg::Ref<HibernatableWebSocketMessageEvent> event)
+      : event(kj::mv(event)) {}
+
+
+  JSG_RESOURCE_TYPE(HibernatableWebSocketMessageController) {
+  }
+
+private:
+  jsg::Ref<HibernatableWebSocketMessageEvent> event;
+};
+
+struct HibernatableWebSocketMessageExportedHandler {
+  // Extension of ExportedHandler covering HibernatableWebSocketMessage handers.
+
+  typedef kj::Promise<void> HibernatableWebSocketMessageHandler(
+      jsg::Ref<HibernatableWebSocketMessageController> controller, jsg::Value env, jsg::Optional<jsg::Ref<ExecutionContext>> ctx);
```

Comment:
The API doesn't call for a "controller" type for this event, and `env` and `ctx` are delivered to the constructor for durable objects.

Take a look at how alarm handlers are specified -- that's a lot closer to what you want here since this is an event type that is only delivered to Durable Objects.